tools/noxfile.py

```diff
@@ -0,0 +1,10 @@
+import nox
+
+nox.options.reuse_existing_virtualenvs = True
+
+
+@nox.session
+def typing(session):
+    session.install("-r", "backups2datalad.req.txt")
+    session.install("mypy", "boto3-stubs[s3]")
+    session.run("mypy", "backups2datalad.py")
```

Comment:
please add running the test(s) and a github actions for typing and tests

Reply:
Done.